ShipMonitor/ShipMonitor.cs

```diff
@@ -1027,12 +1050,16 @@ private void handleJumpedEvent(JumpedEvent @event)
                 if (@event.boostused is null)
                 {
                     Ship ship = GetCurrentShip();
-                    if (@event.fuelused > ship?.maxfuel)
-                    {
-                        ship.maxfuel = @event.fuelused;
-                        ship.maxjump = @event.distance;
-                        if (!@event.fromLoad) { writeShips(); }
-                    }
+
+                    // Get the necessary calculation parameters
+                    ratingConstantFSD.TryGetValue(ship.frameshiftdrive.grade, out decimal ratingConstant);
+                    powerConstantFSD.TryGetValue(ship.frameshiftdrive.@class, out decimal powerConstant);
+                    decimal totalShipMass = ship.unladenmass + ((CargoMonitor)EDDI.Instance.ObtainMonitor("Cargo monitor")).cargoCarried + @event.fuelremaining;
+                    decimal optimalMass = @event.distance * totalShipMass / (decimal)Math.Pow((double)(1000 * @event.fuelused / ratingConstant), 1 / (double)powerConstant);
+
+                    // Max fuel per jump calculated using unladen mass and max jump range w/ just enough fuel to complete max jump
+                    ship.maxfuelperjump = ratingConstant * (decimal)Math.Pow((double)(ship.maxjumprange * (ship.unladenmass + ship.maxfuelperjump) / optimalMass), (double)powerConstant) / 1000;
```

Comment:
I don't think this formula takes Guardian FSD boosters into account? 
Does it need to take the active fuel reservoir into account too?
Perhaps `optimalMass` should be saved as a property of the ship?
I'd also like to suggest encapsulating the calculations for current jump range into a function available to the script resolver or VoiceAttack.
(Note that fuel information can be taken from the Status monitor if needed).

Reply:
For reference, here's the Coriolis calculation: https://github.com/EDCD/coriolis/blob/master/src/app/shipyard/Calculations.js

Reply:
I suppose it can include the reservoir, but that’s why I said “reasonably accurate”.

Can you provide an example where a use case would need it accurate to < 1% ?

Reply:
The only possible example would be extreme exploration but as you say the reservoir in most cases won't have a large impact. If it helps, however, the reservoir fuel is available from the status monitor.